compiler/parser/RdrHsSyn.hs

```diff
@@ -2478,13 +2484,39 @@ argPatOfChoice choiceConName _ = asPatRecWild "arg" choiceConName
 --------------------------------------------------------------------------------
 -- Utilities for constructing types and values
 
+userWrittenTupleName :: RdrName
+userWrittenTupleName = qualifyDesugar $ mkVarOcc "userWrittenTuple"
+
+magicName :: RdrName
+magicName = mkRdrQual (mkModuleName "GHC.Types") $ mkVarOcc "magic"
+
+userWrittenTuple :: LHsExpr GhcPs
+userWrittenTuple =
+  let mkTyStr :: String -> HsType GhcPs
+      mkTyStr lit = HsTyLit noExt (HsStrTy NoSourceText (fsLit lit))
+
+      freeIdentity :: HsType GhcPs
+      freeIdentity =
+        let xVar = mkRdrUnqual (mkTyVarOcc "x")
+            xTyVar = HsTyVar noExt NotPromoted (noLoc xVar)
+        in
+        HsForAllTy
+          noExt
+          [noLoc $ UserTyVar noExt (noLoc xVar)]
+          (noLoc $ HsFunTy noExt (noLoc xTyVar) (noLoc xTyVar))
+
+      appType :: HsExpr GhcPs -> HsType GhcPs -> HsExpr GhcPs
+      appType e t = HsAppType noExt (noLoc e) (HsWC noExt (noLoc t))
```

Comment:
`mkTyStr` can be replaced by the existing `mkSymbol`, and `appType` by `mkAppType`. The only difference is that the existing ones operate on `L`ocated types while yours use the plain types, but that should be a matter of dropping/shifting some `noLoc`s

Reply:
done